test/unit/openapi/streaming/subscription-queue-spec.js

```diff
@@ -11,106 +11,191 @@ describe("openapi SubscriptionQueue", () => {
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_SUBSCRIBE });
+
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_MODIFY_SUBSCRIBE);
             expect(queue.isEmpty()).toBe(true);
         })
 
         it('should merge and only leave subscribe', () => {
-            queue.enqueue(SubscriptionActions.ACTION_UNSUBSCRIBE);
-            queue.enqueue(SubscriptionActions.ACTION_SUBSCRIBE);
-            queue.enqueue(SubscriptionActions.ACTION_UNSUBSCRIBE);
-            queue.enqueue(SubscriptionActions.ACTION_SUBSCRIBE);
+            queue.enqueue({ action: SubscriptionActions.ACTION_UNSUBSCRIBE });
+            queue.enqueue({ action: SubscriptionActions.ACTION_SUBSCRIBE });
+            queue.enqueue({ action: SubscriptionActions.ACTION_UNSUBSCRIBE });
+            queue.enqueue({ action: SubscriptionActions.ACTION_SUBSCRIBE });
+
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_SUBSCRIBE);
+            expect(queue.isEmpty()).toBe(true);
+        })
+
+        it('should not merge patch actions', () => {
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_PATCH });
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_PATCH });
+
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_MODIFY_PATCH);
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_MODIFY_PATCH);
+            expect(queue.isEmpty()).toBe(true);
+        })
 
-            expect(queue.dequeue()).toBe(SubscriptionActions.ACTION_SUBSCRIBE);
+        it('should not insert subscribe after modify subscribe', () => {
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_SUBSCRIBE });
+            queue.enqueue({ action: SubscriptionActions.ACTION_SUBSCRIBE });
+
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_MODIFY_SUBSCRIBE);
+            expect(queue.isEmpty()).toBe(true);
+        })
+
+        it('should remove all actions before unsubscribe followed by modify subscribe', () => {
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_PATCH });
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_PATCH });
+            queue.enqueue({ action: SubscriptionActions.ACTION_UNSUBSCRIBE });
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_SUBSCRIBE });
+
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_UNSUBSCRIBE);
+            expect(queue.dequeue().action).toBe(SubscriptionActions.ACTION_MODIFY_SUBSCRIBE);
+            expect(queue.isEmpty()).toBe(true);
+        })
+
+        it('should insert modify subscribe and remove any non unsubscribe actions', () => {
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_PATCH });
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_PATCH });
+            queue.enqueue({ action: SubscriptionActions.ACTION_MODIFY_SUBSCRIBE });
```

Comment:
this would never happen right? you would have an unsubscribe before this modify...